Core/Converters/JsonIgnoreBadUrlConverter.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using log4net;
+
+namespace CKAN
+{
+    /// <summary>
+    /// Reads a URL string, and returns a Uri object. Returns null if the string
+    /// doesn't parse.
+    /// </summary>
+    public class JsonIgnoreBadUrlConverter : JsonConverter
+    {
+        private static readonly ILog log = LogManager.GetLogger(typeof(JsonIgnoreBadUrlConverter));
+
+        public override object ReadJson(
+            JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            string value = reader.Value == null ? null : reader.Value.ToString();
+
+            if (value == null)
+                return null;
+
+            Uri url;
+
+            try
+            {
+                url = new Uri(value);
+                return url;
```

Comment:
return new Uri(value);
Also why have the declaration outside the try when the catch/following code doesn't use it.